### ISSUE TITLE
Invoke-WebRequest is incredibly slow

### ISSUE THREAD

**Issue by albertvaka** [PowerShell/PowerShell#13414]
## Steps to reproduce

Download any large file, it takes eons to complete:

```powershell
Invoke-WebRequest -O datadog-agent-7-latest.amd64.msi https://s3.amazonaws.com/ddagent-windows-stable/datadog-agent-7-latest.amd64.msi
```

## Expected behavior

I would spect a speed comparable to using a web browser.

## Related issues in SO

https://stackoverflow.com/questions/48268279/powershell-invoke-webrequest-is-consistently-very-slow-how-can-i-fix-this
https://stackoverflow.com/questions/28682642/powershell-why-is-using-invoke-webrequest-much-slower-than-a-browser-download
https://stackoverflow.com/questions/46830703/downloading-large-files-in-windows-command-prompt-powershell

**Comment by albertvaka** [comment]
A workaround seems to be to disable the progress indicator with `$ProgressPreference = 'SilentlyContinue'` which might give a hint toward where the problem is.

**Comment by vexx32** [comment]
What version of PowerShell are you running?

**Comment by albertvaka** [comment]
```
> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.19041.1
```
The Windows 10 default.

**Comment by albertvaka** [comment]
Has this been fixed in more recent versions? Is there any way to upgrade? And... why's Windows not shipping the latest version?

**Comment by vexx32** [comment]
Yeah this was fixed in 6.1 or 6.1 IIRC.

6.x broke quite a few things that PowerShell could previously do just fine on Windows, partly because the project moved from .NET 4.6 to .NET Core. The PowerShell team have been trying to ensure PowerShell ships with the latest .NET Core runtime as much as possible, and Windows release cycles would make this effectively impossible.

And, as mentioned, many things broke initially. The vast majority of those are now not an issue anymore, but some things that used native Windows APIs specifically still could not be ported or open sourced (undocumented APIs, lack of support in .NET, etc.)

TL;DR: Shipping a Windows component and shipping a bleeding-edge open source application are two vastly different worlds, effectively entirely incompatible. And as pwsh is not a full 100% replacement for Windows PowerShell in some remaining cases, it cannot entirely replace it until the original modules are rewritten to work correctly with pwsh. As that code is maintained by its respective product teams, that will likely take some time to come.

Eventually, it may, but at the moment, it's not feasible to ship pwsh in Windows itself the same way Windows PowerShell did. 🙂

**Comment by albertvaka** [comment]
Thanks for the detailed explanation :) It sucks Windows users are stuck with a broken version tough :( Closing the ticket now.

**Comment by rjmholt** [comment]
> Windows users are stuck with a broken version

[Unless you install the newer version](https://github.com/PowerShell/PowerShell/releases/download/v7.0.3/PowerShell-7.0.3-win-x64.msi)
